src/Raven.Client/Documents/BulkInsert/BulkInsertOperation.cs

```diff
@@ -182,6 +182,11 @@ public string Store(object entity)
 
         public async Task StoreAsync(object entity, string id)
         {
+            if (id.EndsWith("/"))
```

Comment:
Check for null id / empty id as well, also invalid.
Move the throws into a dedicate method.